src/app/components/album-details/album-details.service.js

```diff
@@ -0,0 +1,24 @@
+export class AlbumDetailsService {
+  constructor($http) {
+    'ngInject';
+    this.$http = $http;
+  }
+  
+  getAlbum(id) {
+    return this.$http.get(`/album/${id}`)
+            .then(response => response.data)
+            .catch(error => { throw error; });
```

Comment:
Actually is not needed, because I dont have a catch in the controller, and not doing action, but is needed if we make some action in this catch, i.e. $log the error here, or call some other method of this scope. In that case is needed to pass the error to the next catch(of the promise wich consumes the service method, generally in the component). If we don´t do this, or put a return here, the view controller promise that consumes this method will not receive the error in his .catch and and the error goes into its .then. But correct me if I am bad :)